src/security/ldap/src/main/java/org/geoserver/security/ldap/BindingLdapAuthoritiesPopulator.java

```diff
@@ -209,10 +209,12 @@ public Set<GrantedAuthority> getGroupMembershipRoles(
                     // Get ldap groups in form of Pair<String,String> -> Pair<name,dn>
                     final String formattedFilter =
                             MessageFormat.format(groupSearchFilter, userDn, username);
+                    // Replace "\" with "\5C" for search.
+                    String formattedFilterNew = formattedFilter.replace("\\", "\\5C"); 
```

Comment:
The \\5C replacement surprised me at first, but it seems there are even more that might have to be considered:
* https://www.ibm.com/docs/en/db2-data-mgr-console/3.1.x?topic=ldap-requirements-dn-value
* https://ldap.com/ldap-dns-and-rdns/

If that's the case, in addition to adding a test, could you please factor out a utility method doing all the necessary escapes, and then call it from all the places where you did the direct "\\5C" replacement?

Reply:
Hopefully I can quickly realize your proposed addition and change.